lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -1564,10 +1564,15 @@ tokenize_sigil_name([S | T], [], Line, Column, Scope, Tokens) when ?is_upcase(S)
 % If we have an uppercase letter, we keep tokenizing the name.
 tokenize_sigil_name([S | T], NameAcc, Line, Column, Scope, Tokens) when ?is_upcase(S) ->
   tokenize_sigil_name(T, [S | NameAcc], Line, Column + 1, Scope, Tokens);
+% A digit is allowed but an upcase or digit must proceed it.
```

Comment:
I think we should break this in two functions. The first clause will call either tokenize_upper_sigil_name or tokenize_lower_sigil_name. I think it will help remove conditionals. :)

Reply:
You know how I love to minimize conditionals!

Reply:
@josevalim  do you think we should revise the error messages per the change such that it's limited to the potential error that can be encountered? In other words, we would have two different error messages depending on the path. If we are on the lower path, then we only emit an error about how a sigil may starting with lower case may only contain a single lower alpha; if we are on the upper path, we emit an error specific to that?

Reply:
unifying into a single error message is also fine by me